autorest/azure/async.go

```diff
@@ -164,9 +164,17 @@ func (f Future) WaitForCompletion(ctx context.Context, client autorest.Client) e
 // running operation has completed, the provided context is cancelled, or the client's
 // polling duration has been exceeded.  It will retry failed polling attempts based on
 // the retry value defined in the client up to the maximum retry attempts.
-func (f *Future) WaitForCompletionRef(ctx context.Context, client autorest.Client) error {
-	ctx, cancel := context.WithTimeout(ctx, client.PollingDuration)
-	defer cancel()
+func (f *Future) WaitForCompletionRef(inputCtx context.Context, client autorest.Client) error {
+
+	var ctx context.Context
+	if d := client.PollingDuration; d != nil {
```

Comment:
Related to my comment below, suggesting that client.PollingDuration not become a pointer, this would become:
``` Go
var ctx context.Context
if d := client.PollingDuration; d != 0 {
    var cancel context.CancelFunc
    ctx, cancel = context.WithTimeout(inputCtx, d)
    defer cancel()
} else {
    ctx = inputCtx
}
```

Reply:
(updated)